sentry/src/main/java/io/sentry/Hub.java

```diff
@@ -862,4 +862,32 @@ private Scope buildLocalScope(
 
     return null;
   }
+
+  @Override
+  public @NotNull SentryId captureCheckIn(final @NotNull CheckIn checkIn) {
+    return captureCheckInInternal(checkIn);
+  }
+
+  private @NotNull SentryId captureCheckInInternal(final @NotNull CheckIn checkIn) {
```

Comment:
I started out with more overloads but then decided on only using a single `captureCheckIn` method that simply takes a `CheckIn` instead of creating a pyramid of optional params. I'll inline.